gui/src/containers/JobSpec.js

```diff
@@ -52,64 +52,77 @@ const styles = theme => ({
   }
 })
 
-const renderJobSpec = ({classes, jobSpec, jobRunsCount}) => (
-  <Grid container spacing={40}>
-    <Grid item xs={8}>
-      <PaddedCard>
-        <Grid container alignItems='flex-start'>
-          <Grid item xs={6}>
-            <Typography variant='title' className={classes.definitionTitle}>
-              Definition
-            </Typography>
-          </Grid>
-          <Grid item xs={6} align='right'>
-            <CopyJobSpec JobSpec={jobSpecDefinition(jobSpec)} />
-          </Grid>
-          <Grid item xs={12}>
-            <Divider light className={classes.divider} />
-          </Grid>
-          <PrettyJson object={jobSpecDefinition(jobSpec)} />
-        </Grid>
-      </PaddedCard>
-    </Grid>
-    <Grid item xs={4}>
-      <PaddedCard>
-        <Grid container spacing={16}>
-          <Grid item xs={12}>
-            <Typography variant='subheading' color='textSecondary'>ID</Typography>
-            <Typography variant='body1' color='inherit'>
-              {jobSpec.id}
-            </Typography>
-          </Grid>
-          <Grid item xs={12}>
-            <Typography variant='subheading' color='textSecondary'>Created</Typography>
-            <Typography variant='body1' color='inherit'>
-              {jobSpec.createdAt}
-            </Typography>
+const renderJobSpec = ({ classes, jobSpec, jobRunsCount, submitJobSpecRun, fetching, fetchJobSpec }) => {
+  const handleClick = () => {
+    submitJobSpecRun(jobSpec.id).then(() => fetchJobSpec(jobSpec.id))
```

Comment:
Regarding notifications being cleared because of `matchUrl` changing, is it because we're firing two actions here and the second clears the notifications? I'll take a look on your branch.

Reply:
Nvm, it fires `MATCH_ROUTE` regardless

Reply:
Made a PR for this fix onto your branch: https://github.com/NavyAdmiral/chainlink/pull/1

Reply:
I was afraid changing initialstate to state would have lead to duplicate success notifications. Ill check it when I can